### PR TITLE
[master] Fix inconsistent use of args in ssh_auth.managed

### LINKED ISSUE #64442: [BUG] Inconsistent use of args in ssh_auth.managed
**Description**
ssh_auth.managed does not fully respect arguments such as config, fingerprint_hash_type, options, etc

In particular, it will happily use all of the args available when making sure that keys are present, always putting new keys into their correct places. However, when determining what keys to remove or modify, it won't, always using the default `~/.ssh/authorized_keys` file, and other such default options for `ssh.auth_keys` and `ssh_auth.absent` calls

**Setup**
Standard setup in v3005 or v3006. Doesn't matter if it's a VM, on-prem or anything else. It is linux-only, however

Please be as specific as possible and give set-up details.

- [x] on-prem machine
- [x] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [x] used bootstrap to install

**Steps to Reproduce the behavior**
I originally spotted this bug while using a different config file, so!

- Have some keys in `~/.ssh/authorized_keys`
- Make a state with `ssh_auth.managed` and specify a different path for `config`. Use one or more different keys to those you added to the default file
- Apply the state. You'll find your new file is created correctly, but the keys you added to  `~/.ssh/authorized_keys` have been removed
- Swap the keys in the state with some others
- Apply the state again. This time you'll find that the keys originally in the state were never removed, and new keys have only been appended

**Expected behavior**
All functions called by `ssh_auth.managed` should respect arguments provided where applicable

**Screenshots**

**Versions Report**
N/A, exists in v3005 and v3006's code. Haven't checked other releases

**Additional context**

### PR DESCRIPTION
**Pull request by rittycat**
### What does this PR do?
Ensures that arguments are properly used in all functions called by `ssh_auth.managed`. Particularly when it checks for what to remove from an existing config

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/64442

### Previous Behavior
When calling `ssh_auth.absent`, it would not use all arguments given to the state, thus, if you for example specified a custom config file, the state would delete keys from the default config file instead, while only adding new keys to the custom path specified

### New Behavior
Uses all arguments as you would expect in any functions called by the managed state

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/states/ssh_auth.py [hunk 1]
```diff
@@ -580,15 +580,31 @@ def manage(
...
         if __opts__["test"]:
             remove_comment = "{} Key set for removal".format(remove_key)
             ret["comment"] = remove_comment
             ret["result"] = None
         else:
-            remove_comment = absent(remove_key, user)["comment"]
+            remove_comment = absent(
+                remove_key, 
+                user,
+                enc,
```
Comment (Ch3LL) [inline salt/states/ssh_auth.py:600]: Could you pass these in as kwargs, for example `enc=enc`. Just future proofing in case any kwargs's position moves.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will also require test coverage to ensure this doesn't regress in the future.

**Comment by Ch3LL** [comment]
bump @rittycat did you see my review comments?

**Comment by davama** [comment]
thank you for the PR @rittycat

File: tests/pytests/functional/states/test_ssh_auth.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            config=str(new_auth_file),
+            ssh_keys=[""],
+        )
+    with salt.utils.files.fopen(user_ssh_dir / "authorized_keys") as fp:
+        post_data = fp.read()
+
+    assert pre_data == post_data
+    with salt.utils.files.fopen(new_auth_file) as fp:
+        data = fp.read().strip()
+    assert data == file_contents
```
Comment (s0undt3ch) [inline tests/pytests/functional/states/test_ssh_auth.py:60]:
```suggestion


@pytest.fixture(scope="module")
def system_user():
    with pytest.helpers.create_account() as system_account:
        yield system_account
            
            
@pytest.mark.skip_if_not_root
@pytest.mark.destructive_test
@pytest.mark.slow_test
def test_ssh_auth_config(tmp_path, system_user, state_tree):
    """
    test running ssh_auth state when
    different config is set. Ensure
    it does not edit the default config.
    """
    userdetails = system_user.info
    user_ssh_dir = pathlib.Path(userdetails["home"], ".ssh")
    ret = ssh_auth_state.manage(
        name="test",
        user=system_user.username,
        ssh_keys=["ssh-dss AAAAB3NzaCL0sQ9fJ5bYTEyY== root@domain"],
    )

    with salt.utils.files.fopen(user_ssh_dir / "authorized_keys") as fp:
        pre_data = fp.read()

    file_contents = "ssh-dss AAAAB3NzaCL0sQ9fJ5bYTEyY== root@domain"
    new_auth_file = tmp_path / "authorized_keys3"

    with pytest.helpers.temp_file("authorized", file_contents, state_tree):
        ssh_auth_state.manage(
            name="test",
            user=system_user.username,
            source=f"salt://authorized",
            config=str(new_auth_file),
            ssh_keys=[""],
        )
    with salt.utils.files.fopen(user_ssh_dir / "authorized_keys") as fp:
        post_data = fp.read()

    assert pre_data == post_data
    with salt.utils.files.fopen(new_auth_file) as fp:
        data = fp.read().strip()
    assert data == file_contents
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
There's helper code to create/delete temporary accounts.
Also, if the fixture is not needed anywhere else, define the fixture on the module using it.

File: salt/states/ssh_auth.py (hunk 1 above)
Comment (rittycat) [reply salt/states/ssh_auth.py]: Oop. Sorry for never getting back to this, notifications get spammed a lot by private repos. Glad to see this got picked and merged though!

**Pull request closed without merging**